src/hotspot/share/code/codeBlob.cpp

```diff
@@ -315,6 +316,56 @@ AdapterBlob* AdapterBlob::create(CodeBuffer* cb) {
   return blob;
 }
 
+MethodHandleIntrinsicBlob::MethodHandleIntrinsicBlob(
+  Method* method,
+  int mhi_size,
+  CodeBuffer* code_buffer):
+  RuntimeBlob("MethodHandle intrinsic", code_buffer, sizeof(MethodHandleIntrinsicBlob), mhi_size, CodeOffsets::frame_never_safe, 0, NULL),
+  _method(method) {
+  {
+    debug_only(NoSafepointVerifier nsv;)
+    assert_locked_or_safepoint(CodeCache_lock);
+
+    assert(code_buffer->total_oop_size() == 0, "unexpected oop");
+    assert(code_buffer->total_metadata_size() == 0, "unexpected metadata");
+
+    CodeCache::commit(this);
+  }
+}
+
+void* MethodHandleIntrinsicBlob::operator new(size_t s, int size) throw() {
+  return CodeCache::allocate(size, CodeBlobType::NonNMethod);
+}
+
+MethodHandleIntrinsicBlob* MethodHandleIntrinsicBlob::create(const methodHandle& method,
+  CodeBuffer *code_buffer) {
+  code_buffer->finalize_oop_references(method);
```

Comment:
Can you please explain why this is needed? (I'm a bit surprised since the constructor asserts that `total_oop_size` is 0)

Thanks.

Reply:
I don't think finalize_oop_references() makes sense except for nmethods.  If the MH intrinsic could contain contain oops, then GC would need to be able to find and relocate them.